contracts/JBProjects.sol

```diff
@@ -107,31 +103,35 @@ contract JBProjects is JBOperatable, ERC721Votes, Ownable, IJBProjects {
             metadataContentOf[projectId][_metadata.domain] = _metadata.content;
         }
 
-        emit Create(projectId, _owner, _metadata, msg.sender);
+        emit Create(projectId, _owner, _metadata, _msgSender());
     }
 
     /// @notice Allows a project owner to set the project's metadata content for a particular domain namespace.
-    /// @dev Only a project's owner or operator can set its metadata.
+    /// @dev Only a project's controller can set its metadata.
     /// @dev Applications can use the domain namespace as they wish.
     /// @param _projectId The ID of the project who's metadata is being changed.
     /// @param _metadata A struct containing metadata content, and domain within which the metadata applies.
     function setMetadataOf(uint256 _projectId, JBProjectMetadata calldata _metadata)
         external
         override
-        requirePermission(ownerOf(_projectId), _projectId, JBOperations.SET_PROJECT_METADATA)
+        onlyController(_projectId)
```

Comment:
onlyController and dependency on Directory here feels weird, since a project can be created without need for a controller (or directory for that matter). thoughts?

Reply:
what if `metadataContentOf[_projectId][_metadata.domain]` was stored in JBController instead? and set on launchProjectOf, and updated there too.